### PR TITLE
Port DeltaV syndicate objective "teach a lesson"

### PR DESCRIPTION
**Pull request by Ermucat**
## About the PR

Ported https://github.com/DeltaV-Station/Delta-v/pull/2184 from deltaV to harmony(Is ported the right word? I guess it is kinda stealing that might be frowned apon idk)
## Why / Balance

Currently the Kill or Maroon objective is the only way of doing murder within your objectives, but with the limitation that the murdered one must be permanant killed. This system CAN be anti-fun for the victim and the murderer, I dont think it should be removed but providing a alternative I feel is needed
## Technical details

Added new folder to Harmony/Locale/En-us
Added Teachalessoncomponent
added Teachalessonconditionsystem
added things to traitor objective ymls
## Media

https://github.com/user-attachments/assets/83a55be5-c2e6-454d-a914-e7e88d5414e4

## Requirements

- [X] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes

**Changelog**

:cl:
- add: Added new syndicate objective, Teach someone a lesson

### CODE REVIEW

**Review by FluffMe — CHANGES_REQUESTED** [review]
- When porting Delta V or other fork content, please preserve the `DeltaV` namespace as `_DeltaV`.
- Additionally, try to trace any changes to the feature after initial PR and implement them too.
- Additionally, entities from `base_objectives.yml` and `traitor.yml` can be moved to the `_Harmony` and/or `_DeltaV`.

I will do a more thorough review once these points are addressed.

Thank you for working on this! Adding Admin input label while I am at it.

**Comment by spanky-spanky** [comment]
There has been very recent admin discussion of wanting more diverse objectives so the admin consensus will probably be yes.

**Comment by Ermucat** [comment]
@FluffMe 
Could you ellaborate on what preserving the namespace is?

**Comment by FluffMe** [comment]
> @FluffMe 
> Could you ellaborate on what preserving the namespace is?
> 

- Original files are in DeltaV folders. They should be in _DeltaV folders on Harmony.
- c# files have `namespace` declaration. These should also keep DeltaV in the namespace and they mimic the filepaths

**Review by FluffMe — CHANGES_REQUESTED** [review]
Namespaces are very wrong.
Can't review in detail from the phone, but I'll fix it in the next couple of days or on the weekend, unless you figure it out sooner.

Check for harmony C#-containing PRs for reference, if need be.

**Comment by TheCrimsonJupiter** [comment]
Does this replace the current kill/maroon objectives or does it add a new kind of objective for traitors? If it replaces them like they do on DeltaV, I foresee a big rule conflict and general bad vibes. 

There are many traitor tools and methods of killing that are designed to round remove, should this no longer ever be a requirement these sorts of tools will just be rulebreaks if they are used 
"Minor antagonists... should not: Actively round remove people not listed as targets by you or another syndicate." 
from Rule 5: Play Antagonists Responsibly. Should this feature REPLACE kill/maroon, like it does on DeltaV, then using any of these methods would be a rule violation and thats just bad game design to have features that conflict with our philosophy. Similar applies regardless of rules. Should somebody be murdered as a "teach a lesson" target and then fly off into space and be round removed, thats going to be far more upsetting than just being killed as a regular objective for you are now round removed even though you weren't required to be by the game. 

All of this doesn't really apply should it be added as a separate objective, for then discretion and use of tools can be scaled to the task at hand.

**Comment by TheCrimsonJupiter** [comment]
> It does not replace kill and maroon, it is merely a diffreent alternative to add variety

In that case all I wrote was just nonsense rambling. I'm a-okay with this as a new objective, but I think the name should be changed "Teach (name) a lesson" doesn't make any sense. When somebody wakes up from being dead they shouldn't be remembering anything that happened, so there is no lesson being taught here. If there was something with leaving a business card on a dead person so they find it when they wake up maybe that'd make more sense, but that'd take some extra programming stuff that I don't know if anybody is interested in. Something like "Kill them once" would be very basic and not very interesting, but would be less vague and nonsense and get the idea across a million times better.

File: Resources/Prototypes/Objectives/base_objectives.yml [hunk 1]
```diff
@@ -111,3 +111,4 @@
   id: BaseCodeObjective
   components:
   - type: CodeCondition
+
```
Comment (FluffMe) [inline Resources/Prototypes/Objectives/base_objectives.yml:114]: Unnecessary change to upstream file. Please revert.

File: Resources/Prototypes/_Harmony/Objectives/base_objectives.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  components:
+  - type: Objective
+    unique: false
+    icon:
+      sprite: Objects/Weapons/Melee/fireaxe.rsi
+      state: icon
+  - type: ObjectiveBlacklistRequirement
+    blacklist:
+      components:
+      - SocialObjective
```
Comment (FluffMe) [inline Resources/Prototypes/_Harmony/Objectives/base_objectives.yml:15]: This prototype should be declared in `traitor.yml` with `TeachLessonRandomPersonObjective` instead

File: Resources/Prototypes/_Harmony/Objectives/traitor.yml
Comment (FluffMe) [inline Resources/Prototypes/_Harmony/Objectives/traitor.yml:1]: DeltaV content. Wrong location.
Should be `Resources/Prototypes/_DeltaV/Objectives/traitor.yml`

File: Content.Server/_DeltaV/Objectives/Components/Teachalessonconditionststem.cs [hunk 3]
```diff
@@ -0,0 +1,50 @@
+using Content.Server.Objectives.Components;
+using Content.Server.Objectives.Systems;
+using Content.Shared.GameTicking;
+using Content.Shared.Mind;
+using Content.Shared.Objectives.Components;
+using DeltaV.Server.Objectives.Components;
```
Comment (FluffMe) [inline Content.Server/_DeltaV/Objectives/Components/Teachalessonconditionststem.cs:6]: Not needed and wrong. Remove this line

File: Content.Server/_DeltaV/Objectives/Components/Teachalessonconditionststem.cs [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+namespace DeltaV.Content.Server.Objectives.Systems;
```
Comment (FluffMe) [inline Content.Server/_DeltaV/Objectives/Components/Teachalessonconditionststem.cs:8]:
```suggestion
namespace Content.Server.Objectives.Systems;
```

We'll just keep this simple and keep original DeltaV namespaces for this.

File: Content.Server/_DeltaV/Objectives/Systems/TeachAlessonComponent.cs [hunk 5]
```diff
@@ -0,0 +1,9 @@
+using DeltaV.Content.Server.Objectives.Systems;
+
+namespace DeltaV.Server.Objectives.Components;
```
Comment (FluffMe) [inline Content.Server/_DeltaV/Objectives/Systems/TeachAlessonComponent.cs:3]:
```suggestion
using Content.Server.Objectives.Systems;

namespace Content.Server.Objectives.Components;
```

**Review by FluffMe — CHANGES_REQUESTED** [review]
First big bunch of required changes.

After you are done with this you should also implement https://github.com/DeltaV-Station/Delta-v/pull/2412 and test everything properly. Milon's PR also properly sets the DeltaV namespaces.

Trace any other changes or PRs to these files that might be important.

Once this work is done, please test it thoroughly and make sure to keep as much of DeltaV code intact as possible. If you make any changes you must declare them per AGPL licensing. No renaming of the files where it's not needed, etc.

This should be a lot of work for someone with no C# experience, so thank you for tackling this. Feel free to ask questions on Discord if you need help. Experienced contributors should be able to help you.

Good luck!

**Comment by FluffMe** [comment]
Marked as draft to avoid constant test runs on Work In Progress. Undraft when it's done.

**Comment by FluffMe** [comment]
@Ermucat I see that you have requested a review, but the PR is in the draft state.
- Please undraft when it's ready.

File: Resources/Prototypes/_DeltaV/Objectives/traitor.yml [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+  id: TeachLessonRandomPersonObjective
+  description: Kill them, and show everyone we mean business. They only need to die once.
+  components:
+  - type: Objective
+    difficulty: 1.75
+    unique: false
+  - type: TargetObjective
+    title: objective-condition-teach-person-title
+  - type: PickRandomPerson
+  - type: TeachLessonCondition
```
Comment (FluffMe) [inline Resources/Prototypes/_DeltaV/Objectives/traitor.yml:13]: In https://github.com/DeltaV-Station/Delta-v/blob/master/Resources/Prototypes/DeltaV/Objectives/traitor.yml there is no `TeachLessonCondition` component in the `TeachLessonRandomPersonObjective` entity. Which makes sense, as it's defined in one of the parents -- `BaseTeachLessonObjective`.

Is it needed here?

File: Content.Server/DeltaV/Objectives/Components/TeachLessonConditionComponent.cs
Comment (FluffMe) [inline Content.Server/DeltaV/Objectives/Components/TeachLessonConditionComponent.cs:1]: Please relocate to `Content.Server/_DeltaV/Objectives/Components/TeachLessonConditionComponent.cs` (missing `_` in `_DeltaV`)

File: Content.Server/DeltaV/Objectives/Systems/TeachLessonConditionSystem.cs
Comment (FluffMe) [inline Content.Server/DeltaV/Objectives/Systems/TeachLessonConditionSystem.cs:1]: Please relocate to `Content.Server/_DeltaV/Objectives/Systems/TeachLessonConditionSystem.cs` (missing `_` in `_DeltaV`)

File: Resources/Prototypes/_DeltaV/Objectives/traitor.yml [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+# Base
+# teach lesson
+- type: entity
+  abstract: true
+  parent: BaseTargetObjective
+  id: BaseTeachLessonObjective
```
Comment (FluffMe) [inline Resources/Prototypes/_DeltaV/Objectives/traitor.yml:19]: Optional nitpick. Base is better located before the thing that inherits from it.

You can move it above `TeachLessonRandomPersonObjective`.

**Review by FluffMe — CHANGES_REQUESTED** [review]
Please view and implement the requested changes.

As there is a discrepancy between this PR and DeltaV code in the yaml, please review one more time the code in DeltaV [master](https://github.com/DeltaV-Station/Delta-v/) vs this PR for the affected files. For good measure.

Should be the last batch of changes otherwise! Great job!

**Review by FluffMe — APPROVED** [review]

**Pull request merged**